src/warmer.js

```diff
@@ -26,16 +26,17 @@ function addWarmUpFunctionRoleToResources(service, stage, warmerName, warmerConf
     Type: 'AWS::IAM::Role',
     Properties: {
       Path: '/',
-      RoleName: {
+      RoleName: warmerConfig.roleName ? warmerConfig.roleName : {
         'Fn::Join': [
           '-',
-          [
-            service.service,
-            stage,
-            { Ref: 'AWS::Region' },
-            warmerName.toLowerCase(),
-            'role',
-          ],
+          warmerConfig.roleName ? [warmerConfig.roleName]
```

Comment:
This check is not needed.

Either the roleName is used in line 29 or the original auto-generated name should be used.

Please remove this.

Reply:
@L14MJ4M3S0N this can just be:

```
RoleName: warmerConfig.roleName || {
  'Fn::Join': [
    '-', 
    [
      service.service,
      stage,
      { Ref: 'AWS::Region' },
      warmerName.toLowerCase(),
      'role',
    ],
  ],
},
```

Reply:
@juanjoDiaz @icholy 

Fixed. Sorry about the delay, just got back from vacation.